cranelift/codegen/src/opts/icmp.isle

```diff
@@ -65,11 +65,37 @@
              (iconst_u _ 1)))
       extend)
 
+;; Rewrite loose inequalities to strict inequalities:
+;; ule(x, c) == ult(x, c+1), for c != UMAX.
+(rule (simplify (ule (fits_in_64 (ty_int bty)) x (iconst cty (u64_from_imm64 c))))
+      (if-let $false (u64_eq c (ty_umax cty)))
+      (ult bty x (iconst cty (imm64 (u64_add c 1)))))
+
+;; uge(x, c) == ugt(x, c-1), for c != 0.
+(rule (simplify (uge (fits_in_64 (ty_int bty)) x (iconst cty (u64_from_imm64 c))))
+      (if-let $false (u64_eq c 0))
+      (ugt bty x (iconst cty (imm64 (u64_sub c 1)))))
+
+;; sle(x, c) == slt(x, c+1), for c != SMAX.
+(rule (simplify (sle (fits_in_64 (ty_int bty)) x (iconst cty (u64_from_imm64 c))))
+      (if-let $false (u64_eq c (ty_smax cty)))
+      (slt bty x (iconst cty (imm64 (u64_add c 1)))))
+
+;; FIXME: triggers verifier error
+;; sge(x, c) == sgt(x, c-1), for c != SMIN.
+;; (rule (simplify (sge (fits_in_64 (ty_int bty)) x (iconst cty (u64_from_imm64 c))))
+;;      (if-let $false (u64_eq c (ty_smin cty)))
+;;      (sgt bty x (iconst cty (imm64 (u64_sub c 1)))))
+
```

Comment:
Is the main difference to #6130 this commented out rule?

Reply:
Yes

Reply:
Would you mind adding a comment explaining why this is currently commented out? With that added, I think this should be good to merge :+1:

Reply:
I believe it has been fixed by https://github.com/bytecodealliance/wasmtime/commit/0683b84b40207d8ec0c9044ef418fbd1f23a62f9

Reply:
Should it be un-commented now?